src/modm/driver/display/st7586s.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2021, Tomasz Wasilczyk
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+#define MODM_ST7586S_HPP
+
+#include <modm/ui/display/monochrome_graphic_display_horizontal.hpp>
+
+#include "st7586s_protocol.hpp"
+
+namespace modm
+{
+
+template <typename SPI, typename CS, typename RST, typename DC, int Width = 384, int Height = 160>
+// TODO: this controller has pixels packed by 3, not 8 per byte
+class St7586s : public MonochromeGraphicDisplayHorizontal<Width, Height>
+{
+	using Command = impl::st7586s::Command;
+	static constexpr uint8_t pixelsPerByte = 3;
+
+	void sendCommand(Command cmd, const void *data, size_t len);
+	template <typename Data>
+	void sendCommand(Command cmd, Data data) {
+		sendCommand(cmd, &data, sizeof(data));
+	}
+	void sendCommand(Command cmd) {
+		sendCommand(cmd, nullptr, 0);
+	}
+
+	void setWindow(uint16_t x, uint16_t y, uint16_t w, uint16_t h);
```

Comment:
Same method is already called setClipping(...), let's follow the scheme

Reply:
Sure. Ideally setClipping should be a part of the API, not just a naming scheme.